src/keymap-manager.coffee

```diff
@@ -700,14 +716,29 @@ class KeymapManager
     @bindingsToDisable = []
 
   enterPendingState: (pendingPartialMatches, enableTimeout) ->
-    @cancelPendingState() if @pendingStateTimeoutHandle?
+    if @pendingStateTimeoutHandle?
+      @cancelPendingState()
+    else
+      @buildPendingPartialMatchedModiferKeystrokes()
+
     @pendingPartialMatches = pendingPartialMatches
     if enableTimeout
       @pendingStateTimeoutHandle = setTimeout(@terminatePendingState.bind(this, true), @partialMatchTimeout)
 
-  cancelPendingState: ->
+  buildPendingPartialMatchedModiferKeystrokes: ->
```

Comment:
I wonder if it would make more sense to store the matched-keydown-keyup style bindings in a separate array all the time rather than filtering them out of the partial matches array later. You could potentially do this in the  `findMatchCandidates` method.